clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -367,19 +413,25 @@ private void pollSelectionKeys(Iterable<SelectionKey> selectionKeys,
                                 socketChannel.socket().getSendBufferSize(),
                                 socketChannel.socket().getSoTimeout(),
                                 channel.id());
+                        madeReadProgressLastPoll = true;
                     } else
                         continue;
                 }
 
                 /* if channel is not ready finish prepare */
-                if (channel.isConnected() && !channel.ready())
+                if (channel.isConnected() && !channel.ready()) {
                     channel.prepare();
+                    madeReadProgressLastPoll = true;
```

Comment:
Is there a need to set madeReadProgressLastPoll here? It seems we only need to set when doing reads?